x/mongo/driver/auth/oidc.go

```diff
@@ -178,6 +186,53 @@ func (oa *OIDCAuthenticator) providerCallback() (OIDCCallback, error) {
 	return nil, fmt.Errorf("%q %q not supported for MONGODB-OIDC", environmentProp, env)
 }
 
+// getAzureOIDCCallback returns the callback for the Azure Identity Provider.
+func getAzureOIDCCallback(clientID string, resource string, httpClient *http.Client) OIDCCallback {
+	// return the callback parameterized by the clientID and resource, also passing in the user
+	// configured httpClient.
+	return func(ctx context.Context, args *OIDCArgs) (*OIDCCredential, error) {
+		resource = url.QueryEscape(resource)
+		var uri string
+		if clientID != "" {
+			uri = fmt.Sprintf("http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=%s&client_id=%s", resource, clientID)
+		} else {
+			uri = fmt.Sprintf("http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=%s", resource)
+		}
+		req, err := http.NewRequestWithContext(ctx, http.MethodGet, uri, nil)
+		if err != nil {
+			return nil, newAuthError("error creating http request to Azure Identity Provider", err)
+		}
+		req.Header.Add("Metadata", "true")
+		req.Header.Add("Accept", "application/json")
+		resp, err := httpClient.Do(req)
+		if err != nil {
+			return nil, newAuthError("error getting access token from Azure Identity Provider", err)
+		}
+		defer resp.Body.Close()
+		var azureResp struct {
+			AccessToken string `json:"access_token"`
+			ExpiresOn   string `json:"expires_on"`
+		}
```

Comment:
Optional: You can skip the separate integer parsing step by using the `,string` struct tag option for `ExpiresOn`.

E.g.
```go
var azureResp struct {
	AccessToken string `json:"access_token"`
	ExpiresOn   int64  `json:"expires_on,string"`
}
```

Reply:
Oh, cool!